### PR TITLE
Change leftjoin to hashleftjoin

### PR DESCRIPTION
**Pull request by nevermore3**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

#### Description:
`go  from 'Tony Parker' over like WHERE like.likeness > 90 YIELD $$ as dst , edge as e`

BEFORE
image

NOW:
image

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [x] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Review by yixinglu — COMMENTED** [review]
Could you remove the LeftJoin plan node type?

**Comment by Shylock-Hg** [comment]
What's the benefit?

**Comment by nevermore3** [comment]
> Could you remove the LeftJoin plan node type?

ok

**Comment by nevermore3** [comment]
> What's the benefit?



> What's the benefit?

As shown in the figure above, some optimization rules can be applied

File: src/graph/executor/logic/ArgumentExecutor.cpp [hunk 1]
```diff
@@ -22,17 +22,13 @@ folly::Future<Status> ArgumentExecutor::execute() {
...
+  VidHashSet unique;
   for (; iter->valid(); iter->next()) {
     auto &val = iter->getColumn(alias);
     if (val.isNull()) {
       continue;
-    } else if (!val.isVertex()) {
-      return Status::Error("Argument only support vertex, but got %s, which is type %s, ",
-                           val.toString().c_str(),
-                           val.typeName().c_str());
     }
```
Comment (nevermore3) [inline src/graph/executor/logic/ArgumentExecutor.cpp:30]: if (successors_.front()->node()->kind() != PlanNode::Kind::kGetVertices && !val.isVertex()) return Error();

**Review by czpmango — DISMISSED** [review]

**Review by yixinglu — APPROVED** [review]

**Review by czpmango — APPROVED** [review]

**Pull request merged**